metaflow/cmd/diff/__init__.py

```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/env python
+import os
+import shutil
+import sys
+from subprocess import PIPE, run
+from tempfile import TemporaryDirectory
+
+from metaflow._vendor import click
+from metaflow import Run, namespace
+from metaflow.cli import echo_always
+
+EXCLUSIONS = [
+    "metaflow/",
+    "metaflow_extensions/",
+    "INFO",
+    "CONFIG_PARAMETERS",
+    "conda.manifest",
+]
+
+
+def echo(line):
+    echo_always(line, err=True, fg="magenta")
+
+
+def extract_code_package(runspec, exclusions):
+    try:
+        namespace(None)
```

Comment:
nit but to be cleaner, use `_namespace_check=False` in the Run constructor (side-effect free)